oak_tensorflow_service/src/tflite.rs

```diff
@@ -66,51 +70,55 @@ const TENSOR_ARENA_SIZE: usize = 1024;
 #[derive(Default)]
 pub struct TfliteModel {
     tensor_arena: Vec<u8>,
+    output_buffer_len: Option<usize>,
 }
 
 impl TfliteModel {
     pub fn new() -> Self {
         let tensor_arena = vec![0; TENSOR_ARENA_SIZE];
-        Self { tensor_arena }
+        Self {
+            tensor_arena,
+            output_buffer_len: None,
+        }
     }
 
-    pub fn initialize(&self, model_bytes: &[u8]) -> anyhow::Result<()> {
-        let model_bytes_len = model_bytes
-            .len()
-            .try_into()
-            .map_err(|error| anyhow!("Failed to convert usize to u64: {}", error))?;
-        let tensor_arena_len = self
-            .tensor_arena
-            .len()
-            .try_into()
-            .map_err(|error| anyhow!("Failed to convert usize to u64: {}", error))?;
+    pub fn initialize(&mut self, model_bytes: &[u8]) -> anyhow::Result<()> {
+        let model_bytes_len = model_bytes.len();
```

Comment:
Shouldn't this method check if the model was already initialized, and fail if so? In fact I think the TFmodel may be better represented as an enum of either Empty or Initialized (or something like that), like a state machine. Or does that make tracking ownership of the tensor arena harder?

Reply:
Currently this state is represented by the `output_buffer_len: Option<usize>`
I think making the whole structure an enum may be not convenient, since all of our Runtimes are just plain structs and are called like so.

Reply:
Not sure I understand why it would be a problem.

Currently they are just pub structs with non-pub fields, so  nothing externally would care whether they are enums or structs. Having one field (called `output_buffer_len`, which seems quite arbitrary) represent the current state, and keeping the other fields in sync via manually enforced invariants seems problematic in the long term.

Perhaps in another PR though?